Error message

ateucher opened the pull request:
WIP for #164

boshek commented:
Did we want to include the record name in the error message? That would require a bit of work I think and involve re-jigging what gets passed to the `catch_catalogue_error` function. All I have done at this point is add a line to separate the error diagnostics from the error message: 

```
Error: The BC data catalogue is currently unable to process this request
================================================================================
Catalogue request:
  URL: https://openmaps.gov.bc.ca/geo/pub/wfs
  POST fields:
    SERVICE=WFS&VERSION=2.0.0&REQUEST=DescribeFeatureType&typeNames=WHSE_WATER_MANAGEMENT.GW_WATER_WELLS_WRBC_SVW
  Content-Type: application/x-www-form-urlencoded
  Accept-Encoding: gzip, deflate
  Accept: application/json, text/xml, application/xml, */*
  User-Agent: https://github.com/bcgov/bcdata
Catalogue response:
  status: HTTP/1.1 400 
  date: Tue, 04 Feb 2020 16:52:45 GMT
  server: Apache
  x-frame-options: allow-from (null)
  x-control-flow-delay-ms: 0
  content-type: application/xml
  set-cookie: GS_FLOW_CONTROL=GS_CFLOW_258cb0ef:170111d7f4b:-302b
  access-control-allow-origin: (null)
  access-control-allow-credentials: true
  access-control-allow-methods: POST, GET, OPTIONS, HEAD
  access-control-allow-headers: X-Requested-With, Refe
```

ateucher commented:
I think we might need to rename and rethink the scope of that function. It's only used for failed WFS requests isn't it? Possibly we need to create an analogue for failed catalogue requests... then in that one we should include the record, but I don't think we need to for the current WFS-focused function.

ateucher commented:
PS. Love the defining of test sets in `setup.R` that will make life much easier

boshek commented:
It would be some `tryCatch` ugliness but could we catch the catalogue error here:

https://github.com/bcgov/bcdata/blob/c5c29fd7dea91c4cc09c14276b10fe92d10d8902/R/utils.R#L224-L227

with the same function. Then either define a method or even just have some logic to return an informative error? 

Another idea is to re define all the functions in `bcdc_read_functions` as something like `safe_read_csv` and then just check and fail early if the csv file isn't a valid url or is missing.

boshek commented:
@ateucher What do you think about merging this and moving the error handling of the tabular data into a new issue?

ateucher commented:
Yes, great idea. That way at least your fixes for testing (especially the groundwater record) will be merged so we don't keep getting false positives on CI

ateucher commented:
I think this is good to go @boshek - just added a couple more small changes. I renamed `catch_catalogue_error()` to `catch_wfs_error()` and tweaked messages accordingly. I also added a `--run-donttest` flag to one GitHub Actions workflow so those examples continue to be checked via CI.
